cpp/src/io/csv/csv_gpu.cu

```diff
@@ -581,19 +594,37 @@ struct decode_op {
     } else {
       value = decode_value<T>(data, start, end, opts);
     }
+    return true;
+  }
+
+  /**
+   * @brief Dispatch for floating points, which are set to NaN if the input 
+   * is not valid. In such case, the validity mask is set to zero too.
+   */
+   template <typename T,
+             typename std::enable_if_t<std::is_floating_point<T>::value> * = nullptr>
+  __host__ __device__ __forceinline__ bool operator()(
+      const char *data, void *out_buffer, size_t row, long start, long end,
+      ParseOptions const &opts, column_parse::flags flags) {
+    auto &value{static_cast<T *>(out_buffer)[row]};
+
+    value = decode_value<T>(data, start, end, opts);
+    return !std::isnan(value);
```

Comment:
Do we need this `!std::isnan()` check only for floating point values?  What happens if you have an integer `452a29` that would also be invalid?

Reply:
std::isnan() is defined only for floating point types, integers don't have a NaN value.
Additionally, Pandas does not support missing/invalid integer data, only float. The parser actually  throws an error if an integer field has an invalid value.

We can find a way to set the validity mask for ints too, but I don't know if this is desired behavior?

Reply:
Ok, let's just address floats.